rust/stracciatella/src/vfs/android.rs

```diff
@@ -3,25 +3,33 @@
 
 use std::collections::HashSet;
 use std::ffi::CString;
+use std::ffi::OsString;
 use std::fmt;
 use std::io;
 use std::io::SeekFrom;
 use std::path::{Path, PathBuf};
 use std::rc::Rc;
+use std::sync::Mutex;
 
+use lru::LruCache;
 use ndk::asset::{Asset, AssetManager};
 
 use crate::android::get_asset_manager;
 use crate::unicode::Nfc;
 use crate::vfs::{VfsFile, VfsLayer};
 
+/// The size of the cache used for canonicalization
+const CANONICALIZATION_CACHE_SIZE: usize = 256;
```

Comment:
Is this a bit on the low side? 

My thinking is: the strategic map has more than 100 sectors and each visited sector can have several temp files. We can have up to 20 soldiers in team and each soldier have more than 10 BATTLESNDs.

This is text values, so the memory consumption should be nothing when compared to the graphics.

Of course, we could always load a late-game save and measure the actual cache usage.

Reply:
It is actually fine for mobile devices which could have stricter memory limits. So this question should have been on the same line in `dir.rs`

Reply:
This was more or less chosen arbitrarily. I need to check actual cache usage, since I wrote the original code long ago.

Reply:
So looking at the actual cache usage, the current cache size never gets full, even when having all data dumped from the `*.slf` files. This is because it contains directory listings, so each directory within the DirFS only occurs once in the cache.